File: frontend/src/types/interface.ts
export interface InterfaceItem {
  id: number;
  router_id: number;
  interface_name: string;
  interface_type: string;
  rx_bps: number;
  tx_bps: number;
  rx_bytes: number;
  tx_bytes: number;
  rx_packets: number;
  tx_packets: number;
  status: string;
  updated_at: string;
}

File: backend/app/schemas/interface.py
from datetime import datetime

from pydantic import BaseModel


class InterfaceOut(BaseModel):
    id: int
    router_id: int
    interface_name: str
    interface_type: str
    rx_bps: int
    tx_bps: int
    rx_bytes: int
    tx_bytes: int
    rx_packets: int
    tx_packets: int
    status: str
    updated_at: datetime

    class Config:
        from_attributes = True
